node/chainSimulator/send_and_execute.go

```diff
@@ -0,0 +1,73 @@
+package chainSimulator
+
+import (
+	"encoding/hex"
+	"errors"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+)
+
+func (s *simulator) sendTx(tx *transaction.Transaction) (string, error) {
+	shardID := s.GetNodeHandler(0).GetShardCoordinator().ComputeId(tx.SndAddr)
+	err := s.GetNodeHandler(shardID).GetFacadeHandler().ValidateTransaction(tx)
+	if err != nil {
+		return "", err
+	}
+
+	node := s.GetNodeHandler(shardID)
+	txHash, err := core.CalculateHash(node.GetCoreComponents().InternalMarshalizer(), node.GetCoreComponents().Hasher(), tx)
+	if err != nil {
+		return "", err
+	}
+
+	txHashHex := hex.EncodeToString(txHash)
+	_, err = node.GetFacadeHandler().SendBulkTransactions([]*transaction.Transaction{tx})
+	if err != nil {
+		return "", err
+	}
+
+	log.Info("############## send transaction ##############", "txHash", txHashHex)
+
+	return txHashHex, nil
+}
+
+func (s *simulator) SendTxsAndGenerateBlockTilTxIsExecuted(txsToSend []*transaction.Transaction, maxNumOfBlockToGenerateWhenExecutingTx int) ([]*transaction.ApiTransactionResult, error) {
+	hashTxIndex := make(map[string]int)
+	for idx, txToSend := range txsToSend {
```

Comment:
why not calling here directly the node.GetFacadeHandler().SendBulkTransactions([]*transaction.Transaction{tx}) 
That endpoint will give you the list of hashes, directly